### DISCUSSION TITLE
Spread operator in collection read-only property initialization

### DISCUSSION THREAD

**Discussion by Arithmomaniac** [dotnet/csharplang#8336] [Language Ideas]
Suppose the type

```csharp
public class Foo
{
    public IList<int> Bars { get; } = [];
}
```

Using collection [read-only property initialization](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/object-and-collection-initializers#object-initializers-with-collection-read-only-property-initialization), we can do this:

```csharp
var foo = new Foo() 
{ 
    Bars = { 1, 2, 3, 4 }
};
```

But suppose I don't know which values I want at compile time. Then I would be forced to do as follows:

```csharp
// int[] bars = ...;
var foo = new Foo();
foreach (var bar in bars)
{
    foo.Bars.Add(bar);
}
```

I propose that the spread operator be allowed here to do as follows:
```csharp
// int[] bars = ...;
var foo = new Foo()
{
    Bars = { .. bars, 4 }
};
```

**Comment by colejohnson66** [comment]
This could be very useful for `ProcessStartInfo.Arguments`

**Reply by jnm2** [reply]
[`ArgumentList`](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.processstartinfo.argumentlist) rather? One thing you can do is pass `IEnumerable<string> arguments` to the ProcessStartInfo constructor.

**Comment by TahirAhmadov** [comment]
https://github.com/dotnet/csharplang/issues/5354#issuecomment-1439287957

**Reply by Arithmomaniac** [reply]
That would definitely scratch this itch.

I'm personally partial to reserving `[]` unambiguously for _setting_ collection properties and `{}` for _filling_ them. But YMMV.
